src/Sparrow/Json/ArenaMemoryAllocator.cs

```diff
@@ -330,19 +338,27 @@ public void Dispose()
 
                 GC.SuppressFinalize(this);
             }
+            finally
+            {
+                if (disposing)
+                    Monitor.Exit(this);
+            }            
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void Return(AllocatedMemoryData allocation)
         {
             if (_isDisposed)
                 return;
+            GC.SuppressFinalize(allocation);
```

Comment:
Why `AllocatedMemoryData ` have a finalizer?

Reply:
Because it holds buffers and it's main unmamanged memory that it should release (look at the Dispose code)

Reply:
It doesn't have one, removing this